run_list_item.go

```diff
@@ -0,0 +1,86 @@
+package chef
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const (
+	QualifiedRecipe            string = `^recipe\[([^\]@]+)(@([0-9]+(\.[0-9]+){1,2}))?\]$`
+	QualifiedRole              string = `^role\[([^\]]+)\]$`
+	VersionedUnqualifiedRecipe string = `^([^@]+)(@([0-9]+(\.[0-9]+){1,2}))$`
+	FalseFriend                string = `[\[\]]`
+)
+
+var (
+	QualifiedRecipeRegexp            *regexp.Regexp = regexp.MustCompile(QualifiedRecipe)
+	QualifiedRoleRegexp              *regexp.Regexp = regexp.MustCompile(QualifiedRole)
+	VersionedUnqualifiedRecipeRegexp *regexp.Regexp = regexp.MustCompile(VersionedUnqualifiedRecipe)
+	FalseFriendRegexp                *regexp.Regexp = regexp.MustCompile(FalseFriend)
+)
+
+type RunListItem struct {
```

Comment:
We should have godoc comments on the functions and exported structures.

Reply:
Updated!